### ISSUE TITLE
PageFaultsPerSecondStatistic is not taking care of MacOS

### ISSUE THREAD

**Issue by DigitalBox98** [Dawn-of-Light/DOLSharp#487]
When starting the server on MacOS, the below error is raised : 

[21:08:52,652] System.IO.DirectoryNotFoundException: Could not find a part of the path '/proc/stat'.

Looking at  PageFaultStatistics.cs, it only handles Windows else it supposes it's Linux whereas it could be also MacOS : 

public PageFaultsPerSecondStatistic()
        {
            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
            {
                performanceStatistic = new PerformanceCounterStatistic("Memory", "Pages/sec", null);
            }
            else
            {
                performanceStatistic = new LinuxPageFaultsPerSecondStatistic();
            }
        }

**Comment by NetDwarf** [comment]
Heh, yeah I pretty much ignored MacOS :sweat_smile:.  The abstraction was actually mostly to get rid of that error, while also giving *some* info. Of course MacOS was not a consideration, but it is if course possible to implement that as well.

What would be your plan for MacOS?

**Comment by DigitalBox98** [comment]
I know that the command on MacOS exists : sysctl -a

However for my information what is the aim of that code on other systems ?

**Comment by NetDwarf** [comment]
Sorry, forgot to answer, because I had no good one I guess :sweat_smile: 

I guess it should either give information or nothing.

**Comment by DigitalBox98** [comment]
Ok I've checked and now I understood. 
A minimal version is to put dummy figures in order to avoid the errors when the statistic are displayed : CPU=0,0%  DOL=0,0%  pg/s=0,0  dsk/s=0,0

But it should be possible to have that kind of information : 

for total CPU usage : top -l 1 | grep "CPU usage"
=> CPU usage: 2.30% user, 7.11% sys, 90.58% idle 

for DOL usage : ps | grep DOLServer.dll | grep dotnet  
=> 13156 ttys000    1:35.06 dotnet /Users/DOLSharp/Debug/DOLServer.dll
then top | grep 13156
=> PID    COMMAND      %CPU TIME     #TH    #WQ  #PORT MEM    PURG   CMPRS  PGRP  PPID  STATE    BOOSTS         %CPU_ME %CPU_OTHRS UID  FAULTS   COW   MSGSENT
13156  dotnet           9.1  01:41.23 22     1   84     1232M  0B    0B    13152 13152 sleeping *0[1]      0.00000 0.00000    501 350119   793   57429     28695     5452334+   287346+   2683414+  27      1313238+ 96.8  73964492  282739728 180    nicolas               N/A    N/A   N/A   N/A   N/A   N/A  
(not sure which value from this one as 9.1% is not the accurate value)

for page faults : vm_stat | grep "faults"    
=> "Translation faults":                 127510514.

for disks : iostat -d  
=>              disk0 
    KB/t  tps  MB/s 
   24.40   15  0.36 

Each type of information need to reprocess a little bit the output of the commands to retrieve the value

**Comment by NetDwarf** [comment]
Yeah with Linux this would be possible in a similar manner, but I didn't want to execute external commands, if I could avoid it. MacOS might not have that option (but I really don't know).

For the exact values I guess "translation faults" is something different. The 9.1% CPU value might actually be correct, if it is the utilization equivalent of a single thread i.e. if a 4 thread CPU can have 400% utilization total.

To be frank it would be totally okay, if you are not displaying some values for this OS or add others that seem to be of interest to you.
